oximeter/db/src/client/mod.rs

```diff
@@ -960,6 +984,125 @@ impl Client {
         }
         Ok(())
     }
+
+    /// Given a list of timeseries by name, delete their schema and any
+    /// associated data records from all tables.
+    async fn expunge_timeseries_by_name(
+        &self,
+        replicated: bool,
+        to_delete: &[TimeseriesName],
+    ) -> Result<(), Error> {
+        // The version table should not have any matching data, but let's avoid
+        // it entirely anyway.
+        let tables = self
+            .list_oximeter_database_tables(ListDetails {
+                include_version: false,
+                replicated,
+            })
+            .await?;
+
+        const DELETE_BATCH_SIZE: usize = 1000;
+        let maybe_on_cluster = if replicated {
+            format!("ON CLUSTER {}", crate::CLUSTER_NAME)
+        } else {
+            String::new()
+        };
+        for chunk in to_delete.chunks(DELETE_BATCH_SIZE) {
+            let names = chunk
+                .iter()
+                .map(|name| format!("'{name}'"))
+                .collect::<Vec<_>>()
+                .join(",");
+            debug!(
+                self.log,
+                "deleting chunk of timeseries";
+                "timeseries_names" => &names,
+                "n_timeseries" => chunk.len(),
+            );
+            for table in tables.iter() {
+                let sql = format!(
+                    "ALTER TABLE {}.{} \
+                    {} \
+                    DELETE WHERE timeseries_name in ({})",
```

Comment:
Maybe we're using the ALTER/DELETE (heavyweight) method due to the possibility of https://clickhouse.com/docs/en/sql-reference/statements/delete#performance-considerations-when-using-lightweight-delete? This only happens on schema upgrade, so would the ["lightweight"](https://clickhouse.com/docs/en/sql-reference/statements/delete) be an alternative since we don't care about the table anymore or predictable time for its expungement? Just bringing this "lightweight" version as a possibility (though it's only available for *MergeTree table engine families*). Additionally, that post mentions that the `TRUNCATE TABLE` command could be more useful for full table removals.

Reply:
I don't have much experience with any form of the deletes, so I opted for the most "aggressive" form. I don't think performance considerations are that important here, since this is currently only done during offline schema upgrades. There are no select queries to interfere with. I'd prefer to "truly" delete all the data from disk, which I _think_ this approach does based on [this section](https://clickhouse.com/docs/en/sql-reference/statements/alter#synchronicity-of-alter-queries).

> Additionally, that post mentions that the TRUNCATE TABLE command could be more useful for full table removals.

We're not removing any tables, only records inside them.

Reply:
On the "aggressive" form, that's what I figured you were going for. It def does delete at the operation start. 

On `TRUNCATE TABLE`, I thought it was just removing the records/data in the table and not the table itself? But, I think the aggressive form on schema-upgrade makes sense.

Reply:
Yeah I didn't phrase that well. I meant to say that we're removing _specific_ records, not all of them. My bad.